odl/test/space/pspace_test.py

```diff
@@ -900,5 +900,110 @@ def test_reductions():
     assert x.ufuncs.max() == 3.0
 
 
+def test_array_wrap_method():
+    """Verify that the __array_wrap__ method for NumPy works."""
+    space = odl.ProductSpace(odl.uniform_discr(0, 1, 10), 2)
+    x_arr, x = noise_elements(space)
+    y_arr = np.sin(x_arr)
+    y = np.sin(x)  # Should yield again an ODL product space element
+
+    assert y in space
+    assert all_equal(y, y_arr)
+
+
+def test_real_imag_and_conj():
+    """Verify that .real .imag and .conj() work for product space elements."""
+    space = odl.ProductSpace(odl.uniform_discr(0, 1, 3, dtype=complex),
+                             odl.cn(2))
+    x = noise_element(space)
+
+    # Test real
+    expected_result = space.real_space.element([x[0].real, x[1].real])
+    assert x.real == expected_result
+
+    # Test imag
+    expected_result = space.real_space.element([x[0].imag, x[1].imag])
+    assert x.imag == expected_result
+
+    # Test conj. Note that ProductSpace does not implement asarray if
+    # is_power_space is false. Hence the construction below
+    expected_result = space.element([x[0].conj(), x[1].conj()])
+    x_conj = x.conj()
+    assert x_conj[0] == expected_result[0]
+    assert x_conj[1] == expected_result[1]
+
+
+def test_real_setter():
+    """Verify that the setter for the real part of an element works."""
+    space = odl.ProductSpace(odl.uniform_discr(0, 1, 3, dtype=complex),
+                             odl.cn(2))
+    x = noise_element(space)
+    x_imag = x.imag
+
+    # Test for element in ProductSpace where imaginary part is zero
+    tmp = noise_element(space)
+    newreal = space.element(tmp.real)
+    x.real = newreal
+    assert x.real == newreal.real
+
+    # Test for element in ProductSpace with none-zero imaginary part
```

Comment:
TBH, I wouldn't even test with complex right-hand sides. It's ultimately up to the backend (here Numpy) how to handle this, we don't really define ourselves what should happen in this case.